src/read.c

```diff
@@ -485,7 +485,9 @@ void ReadCallVarAss (
     /* get the variable as a global variable                               */
     if ( type == ' ' ) {
         type = 'g';
-        var = GVarName( TLS(Value) );
+        /* we do not want to call GVarName on this value until after we
+         * have checked if this is the argument to a lambda function       */
+        strcpy(varname, TLS(Value));
```

Comment:
Perhaps I am paranoid, but shouldn't this use strlcpy or strxcpy? Alas, it's a minor details (i.e. in my eyes not a reason against merging this as-is)